contracts/validators/ConstantMultipleValidator.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// BaseMPACommand.sol
+
+// Copyright (C) 2021-2021 Oazo Apps Limited
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.0;
+
+import { RatioUtils } from "../libs/RatioUtils.sol";
+import { IValidator } from "../interfaces/IValidator.sol";
+import { ServiceRegistry } from "../ServiceRegistry.sol";
+
+contract ConstantMultipleValidator is IValidator {
+    using RatioUtils for uint256;
+
+    ServiceRegistry public immutable serviceRegistry;
+
+    constructor(ServiceRegistry _serviceRegistry) {
+        serviceRegistry = _serviceRegistry;
+    }
+
+    function decode(bytes[] memory triggersData)
+        public
+        pure
+        returns (uint256[] memory cdpIds, uint256[] memory triggerTypes)
+    {
+        uint256[] memory _cdpIds = new uint256[](triggersData.length);
+        uint256[] memory _triggerTypes = new uint256[](triggersData.length);
+        for (uint256 i = 0; i < triggersData.length; i += 1) {
+            (_cdpIds[i], _triggerTypes[i]) = abi.decode(triggersData[i], (uint256, uint16));
+        }
+
+        return (_cdpIds, _triggerTypes);
+    }
+
+    function validate(uint256[] memory replacedTriggerId, bytes[] memory triggersData)
+        external
+        pure
+        returns (bool)
+    {
+        (uint256[] memory cdpIds, uint256[] memory triggerTypes) = decode(triggersData);
+        require(triggersData.length == 2, "validator/wrong-trigger-count");
+        require(triggerTypes[0] == 3 && triggerTypes[1] == 4, "validator/wrong-trigger-type");
+        for (uint256 i = 0; i < triggersData.length - 1; i++) {
```

Comment:
why `triggersData.length - 1`?

Reply:
`triggersData.length - 1` is only valid to check if all cdpIds are the same (because `cdpIds[i + 1]`), so in fact for BS this loop will be exetuted only once and compare cdpId[0] with cdpId[1].
I'll probably just change it to `require(cdpIds[0] == cdpIds[1], "validator/different-cdpids");` and remove the loop. I was thinking about bigger groups of triggers, but that's not the case for bs.

Reply:
yup, no need to do a loop here since Constant Multiple validator will be used w/ 2 triggers only

Reply:
nit: `validator/different-cdps`